src/foam/core/FObject.java

```diff
@@ -424,12 +425,18 @@ default int compareTo(Object o) {
   }
 
   default void validate(foam.core.X x) {
-
     List<PropertyInfo> props = getClassInfo().getAxiomsByClass(PropertyInfo.class);
+    List<String> exceptions = new ArrayList<String>();
     for ( PropertyInfo prop : props ) {
-      prop.validateObj(x, this);
+      try {
+        prop.validateObj(x, this);
+      } catch (IllegalStateException e) {
+        exceptions.add(prop.getName() + ": " + e.getMessage());
```

Comment:
So here we would look for CompoundException in the context and add to it. 
or create it on 429.  But a CompoundException in the context would capture the model plus any nested fobjects... as well.